src/wp-content/themes/twentysixteen/js/functions.js

```diff
@@ -4,210 +4,290 @@
  *
  * Contains handlers for navigation and widget area.
  */
+ ( function () {
+	'use strict';
 
-( function( $ ) {
-	var body, masthead, menuToggle, siteNavigation, socialNavigation, siteHeaderMenu, resizeTimer;
+	var masthead, menuToggle, siteNavigation, socialNavigation, siteHeaderMenu, resizeTimer;
+
+	function matches( el, sel ) {
+		if ( Element.prototype.matches ) {
+			return el.matches( sel );
+		}
+
+		if ( Element.prototype.msMatchesSelector ) {
+			return el.msMatchesSelector( sel );
+		}
+	}
+
+	// Based on https://developer.mozilla.org/en-US/docs/Web/API/Element/closest#Polyfill
+	function closest( el, sel ) {
+		if ( el.closest ) {
+			return el.closest( sel );
+		}
+
+		var current = el;
+
+		do {
+			if ( matches( el, sel ) ) {
+				return current;
+			}
+			current = current.parentElement || current.parentNode;
+		} while ( current !== null && current.nodeType === 1 );
+
+		return null;
+	}
+
+	function forEachNode( parent, selector, fn ) {
+		var nodes = parent.querySelectorAll( selector );
+		for ( var i = 0; i < nodes.length; i++ ) {
+			fn( nodes[ i ] );
+		}
+	}
 
 	function initMainNavigation( container ) {
+		forEachNode( container, '.menu-item-has-children > a', function ( el ) {
```

Comment:
Per https://github.com/WordPress/wordpress-develop/pull/2297#issuecomment-1041128438, we may need to add a check before here for whether `container` is null.